### DISCUSSION TITLE
dataTable default column filtering

### DISCUSSION THREAD

**Discussion by Amit-ga** [primefaces/community#1073] [PrimeReact] [answered]
Hello, 
I have searched a lot and couldn't find a definite answer, I would really appreciate your help.
Can I define a default column filtering for each column via code, so when the table first loads it already appears with the filters?
(I already defined filters for each column, I just want to add a default value for each filters for when the table loads?)
Thank you.

**Comment by melloware** [comment, marked as answer]
Yes. See this code: https://github.com/melloware/quarkus-primereact/blob/2dc7693594062ff6696b8e3eb7b8ad0a4853fd06/src/main/webui/src/CrudPage.tsx#L52-L68

instead of `value: ''` just put the value you want in that column filter by default?

**Comment by melloware** [comment]
Did you forget `dataType="date"` on your column?

```xml
<Column
					field="completionDate"
					header="Date Completed"
					sortable
					filter
					filterElement={dateFilterTemplate}
					body={completionDateBodyTemplate}
					dataType="date"
				/>
```
